### PR TITLE
feat: add channels support

### LINKED ISSUE #39: Join/create channel

### LINKED ISSUE #40: Leave channel

### LINKED ISSUE #41: Search channels
- [ ] Review integration with kernel (decentraland/kernel#328)

### LINKED ISSUE #42: Get channels joined/friends
- Check if pagination can be added here

### PR DESCRIPTION
**Pull request by jmoguilevsky**
Closes #39 
Closes #40 
Closes #41
Closes #42

### CODE REVIEW

File: src/MessagingClient.ts [hunk 1]
```diff
@@ -284,6 +290,76 @@ export class MessagingClient implements MessagingAPI {
...
+                creation_content: {
+                    type: CHANNEL_TYPE
+                }
+            })
+            return {
+                id: room_id,
+                type: ConversationType.CHANNEL
+            }
+        } catch (error) {
+            throw error
```
Comment (moliva) [inline src/MessagingClient.ts:315]: Let's not do a catch and rethrow if we are not doing anything else with the error here.

File: src/MessagingClient.ts [hunk 2]
```diff
@@ -284,6 +290,76 @@ export class MessagingClient implements MessagingAPI {
...
+                    is_direct: false,
+                    visibility: Visibility.Public,
+                    creation_content: {
+                        type: CHANNEL_TYPE
+                    }
+                }
+            )
+            return conversation
+        } catch (error) {
+            throw error
```
Comment (moliva) [inline src/MessagingClient.ts:342]: Same as above.

File: src/MessagingClient.ts [hunk 3]
```diff
@@ -284,6 +290,76 @@ export class MessagingClient implements MessagingAPI {
...
+        }
+    }
+
+    async leaveChannel(roomId: string): Promise<void> {
+        try {
+            await this.matrixClient.leave(roomId)
+        } catch (error) {
+            throw error
```
Comment (moliva) [inline src/MessagingClient.ts:359]: Same consideration about errors.

File: src/MessagingClient.ts [hunk 4]
```diff
@@ -284,6 +290,76 @@ export class MessagingClient implements MessagingAPI {
         return conversation
     }
 
+    /** Get or create a channel with the given users
+     * If the channel already exists this will return the channel and won't invite the passed ids
+     * If the channel is created, all user ids will be invited to join
+     */
+    async createChannel(channelName: string, userIds: SocialId[]): Promise<Conversation> {
+        try {
+            const { room_id } = await this.matrixClient.createRoom({
```
Comment (moliva) [inline src/MessagingClient.ts:299]: Does it make sense to reuse the `getOrCreateConversation` method here instead?

Comment (moliva) [reply src/MessagingClient.ts:299]: Also, thinking if we should validate that the channel name is not a DM name instead.

File: src/MessagingClient.ts (hunk 1 above)
Comment (jmoguilevsky) [reply src/MessagingClient.ts:315]: This is the reason why this is still in draft, still syncing with juli on how we'll handle/communicate this errors

File: src/MessagingClient.ts (hunk 4 above)
Comment (guidota) [reply src/MessagingClient.ts]: removed this method and validate channel name with a regex

**Review by moliva — APPROVED** [review]

**Pull request merged**